src/Mooc/Courses/Domain/Course.php

```diff
@@ -12,17 +12,20 @@ final class Course extends AggregateRoot
     private CourseId       $id;
     private CourseName     $name;
     private CourseDuration $duration;
+    private CourseCreatedTime $createdAt;
 
-    public function __construct(CourseId $id, CourseName $name, CourseDuration $duration)
+    public function __construct(CourseId $id, CourseName $name, CourseDuration $duration, CourseCreatedTime $createdAt)
     {
         $this->id       = $id;
         $this->name     = $name;
         $this->duration = $duration;
+        $this->createdAt = $createdAt;
     }
 
     public static function create(CourseId $id, CourseName $name, CourseDuration $duration): self
     {
-        $course = new self($id, $name, $duration);
+        $createdAt = new CourseCreatedTime(new \DateTimeImmutable());
```

Comment:
If you want to test it better, instead of doing the `new \DateTimeImmutable()`, you could send this to this method and in the AS call a `clock`